src/rgw/rgw_zone.h

```diff
@@ -1421,4 +1420,27 @@ class RGWPeriod
 };
 WRITE_CLASS_ENCODER(RGWPeriod)
 
+
+namespace rgw::sal { class ConfigStore; }
+
+// global state about the multisite configuration. initialized once during
+// startup and may be reinitialized by RGWRealmReloader, but is otherwise
+// immutable at runtime
+class RGWZoneConfig {
+  RGWRealm realm;
+  RGWPeriod period;
+  RGWZoneParams zone_params;
+  RGWZoneGroup* this_zonegroup = nullptr; // -> period.period_map.zonegroups
+  RGWZone* this_zone = nullptr; // -> this_zonegroup->zones
+ public:
+  int load(const DoutPrefixProvider* dpp, optional_yield y,
+           rgw::sal::ConfigStore* store);
+
+  const RGWRealm& get_realm() const { return realm; }
+  const RGWPeriod& get_period() const { return period; }
+  const RGWZoneGroup& get_this_zonegroup() const { return *this_zonegroup; }
+  const RGWZone& get_this_zone() const { return *this_zone; }
```

Comment:
These all assume that the relevant classes no longer inherit from RGWSystemMetaObj, right?

Reply:
yeah, that RGWSystemMetaObj refactoring would be a prerequisite

Reply:
what i've done so far is to turn `RGWRealm`/`RGWZoneGroup`/`RGWZone`/etc into raw structs, and move `RGWSystemMetaObj` and its subclasses into `rgw_zone_rados.*` with names like `RGWRealmObj`/`RGWZoneGroupObj`/`RGWZoneObj`/etc

this doesn't help us remove any casts though, it just replaces existing uses with the new rados-specific versions

we might instead try to implement `RGWRealmObj` and friends in terms of this `rgw::sal::ConfigStore` instead of leaving them rados-specific - but i think i'd rather just nuke everything related to `RGWSystemMetaObj` and code against the `rgw::sal::ConfigStore` API directly

Reply:
This sounds good.